messaging/src/main/java/org/axonframework/messaging/annotation/HandlerComparator.java

```diff
@@ -40,21 +43,20 @@ public final class HandlerComparator {
     private static final Comparator<MessageHandlingMember<?>> INSTANCE =
             Comparator.comparingInt((ToIntFunction<MessageHandlingMember<?>>) MessageHandlingMember::priority)
                       .reversed()
-                      .thenComparing(
-                              (Function<MessageHandlingMember<?>, Class<?>>) MessageHandlingMember::payloadType,
-                              HandlerComparator::compareHierarchy
-                      )
-                      .thenComparing(Comparator.comparingInt(
-                              (ToIntFunction<MessageHandlingMember<?>>) m -> m.unwrap(Executable.class)
-                                                                              .map(Executable::getParameterCount)
-                                                                              .orElse(1)
-                                     ).reversed()
-                      )
-                      .thenComparing(
-                              m -> m.unwrap(Executable.class)
-                                    .map(Executable::toGenericString)
-                                    .orElse(m.toString())
-                      );
+                      .thenComparing((memberOne, memberTwo) -> {
+                          Comparator<MessageHandlingMember<?>> handlerComparator =
+                                  Comparator.comparing(
+                                                    (Function<MessageHandlingMember<?>, Class<?>>) MessageHandlingMember::payloadType,
+                                                    HandlerComparator::compareHierarchy
+                                            )
+                                            .thenComparing(Comparator.comparingInt(HandlerComparator::parameterCount)
+                                                                     .reversed())
+                                            .thenComparing(HandlerComparator::executableName);
+
+                          return memberOne instanceof ReversedOrder && memberTwo instanceof ReversedOrder
```

Comment:
Unfortunately, this approach is non-deterministic. It may result in `a > b`, `b > c` but then `c < a`. The order of the sorted elements highly depends on the order they were detected.
Instead, I think we should give interceptors only acting on results the higher priority (and thus lower order), so they intercept the messages first, meaning they operate on the result last, by default.

Reply:
You got a point there, bummer... There's one thing that's not clear to me with your suggestion, how would that solve the ordering between result handling interceptors, which should be reversed compared to the "regular" approach?

Reply:
After some offline discussions, we agree that by adjusting the `ResultHandlingInterceptorMember#priority`, we can further reduce the chance described above.

Although, in theory, one might still end up with an undesirable ordering by setting the priority to `Integer.MIN_VALUE`, the chances are slim because:
1. Someone needs to implement a `MessageHandlingMember` with the same priority.
2. Someone needs to combine this custom `MessageHandlingMember` with  `ResultHandlingInterceptorMembers`.
3. Several of both should be present to further increase the probability of an incorrect order.

Hence, in practice, we feel the provided solution *with* and adjusted priority for `ResultHandlingInterceptorMember` is sufficient.